### ISSUE TITLE
Add setup-pre-commit-rs action

### ISSUE THREAD

**Issue by j178** [j178/pre-commit-rs#109]

**Comment by j178** [comment]
This action should also handle tools cache, like `uv`, because venvs created by uv use symlink by default, if restoring from cache without uv cache, those venvs are invalid.

**Comment by ulgens** [comment]
+1 on this. 

Coming from https://github.com/j178/prek/issues/621, I checked how other people / projects handle this need: https://github.com/search?q=prek+%28path%3A*.yaml+OR+path%3A*.yml%29+path%3Aworkflows%2F&type=Code&ref=advsearch&l=&l= It looks like there are various approaches, but none of them looked ideal to me personally. 

This is the config I'm currenly using: https://github.com/ulgens/django-blasphemy/blob/aec8e1218f0723f0152366634744c5a0bd10f0ea/.github/workflows/prek.yml but I adapted that file from a previously existing pre-commit setup. If there are any redundant or missing optimizations, please let me know and I'd like to optimize that config to its best version and contribute back to the project.

**Comment by j178** [comment]
Hey @ulgens! Just wanted to let you know we now have a dedicated GitHub Action for installing and running prek: https://github.com/j178/prek-action. If you have any ideas for improvements, that’s the best place to contribute!

**Comment by ulgens** [comment]
Hey @j178, thanks for the info. I checked the repo and I have couple of questions:
* https://github.com/j178/prek-action/pull/1#issuecomment-3297444627 uv looked like the best choice, I couldn't catch why it was reverted. (thanks for the quick response on the issue, I wasn't able to complete this comment before you replied 😄 )
* Would that action need a bit more nuanced caching? Is it really enough just to check hooks config but not prek or underlying Python versions?
* Would you consider allowing to override Python and prek versions?

**Comment by ulgens** [comment]
Btw, is there any project actively using the action yet? I'd like to compare its performance with alternative approaches.

**Comment by j178** [comment]
> Would that action need a bit more nuanced caching? Is it really enough just to check hooks config but not prek or underlying Python versions?

Yeah, I think we definitely need to add at least the host architecture and platform to the cache.

For prek version - Instead of using the prek version directly in the cache key, we should add a cache version. We’d only bump the cache version when there are breaking changes in prek - that way we don’t invalidate the cache with every prek update. For now, we use `prek-v1-` for this purpose:

https://github.com/j178/prek-action/blob/2c4d2cc7e65d49ca5e22aeaf29b9e1996893f987/action.yaml#L17

For underlying Python versions - I don’t think so. Python is only used to create the hook virtualenv, and if it’s not suitable for a particular hook, it’ll just get recreated anyway. prek is a framework for multiple languages, Python and uv are not that special in prek.

> Would you consider allowing to override Python and prek versions?

Yep, definitely.

**Comment by j178** [comment]
> Btw, is there any project actively using the action yet?

I don’t think so - you’d probably have to create one.
